liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -354,19 +355,54 @@ public void runTestThread(ThreadPoolExecutor executor, String regexp, File logFi
         @Override
         public void checkConfigFile(File configFile){
             try {
-                List<String> configFeatures = getConfigFeatures(configFile);
-                configFeatures.removeAll(this.existingConfigFeatures);
-                if (!configFeatures.isEmpty()) {
+                ServerFeature servUtil = new ServerFeature();
+                log.info("serverDirectory: " + serverDirectory + "; exists: " + serverDirectory.exists());
+                Set<String> features = servUtil.getServerFeatures(serverDirectory);
+                log.info("features: " + features.size());
+                if (!features.isEmpty()){
+                    List<String> configFeatures = new ArrayList<String>(features);
                     log.info("Configuration features have been added");
                     runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-feature", serverName,
                             configFeatures);
                     this.existingConfigFeatures.addAll(configFeatures);
+                    ListIterator<String> iterator = this.existingConfigFeatures.listIterator();
+                    while (iterator.hasNext()) {
+                        iterator.set(iterator.next().toLowerCase());
+                    }
+                    // convert to set to remove duplicate features
+                    Set<String> set = new HashSet<String>(this.existingConfigFeatures);
+                    this.existingConfigFeatures.clear();
+                    this.existingConfigFeatures.addAll(set);
+                    for (String test : this.existingConfigFeatures){
+                        log.info("existing: " + test);
+                    }
                 }
             } catch (Exception e) {
                 log.debug("Failed to read configuration file", e);
             }   
         }
         
+        public boolean compile(File dir) {
+            try {
+                if (dir.equals(sourceDirectory)) {
+                    log.info("Running maven-compiler-plugin:compile");
+                    runMojo("org.apache.maven.plugins:maven-compiler-plugin", "compile", null, null);
+                    log.info("Running maven-compiler-plugin:resources");
+                    runMojo("org.apache.maven.plugins:maven-resources-plugin", "resources", null, null);
+                }
+                if (dir.equals(testSourceDirectory)) {
+                    log.info("Running maven-compiler-plugin:testCompile");
+                    runMojo("org.apache.maven.plugins:maven-compiler-plugin", "testCompile", null, null);
+                    log.info("Running maven-compiler-plugin:testResources");
+                    runMojo("org.apache.maven.plugins:maven-resources-plugin", "testResources", null, null);
+                }
+                return true;
+            } catch (Exception e) {
```

Comment:
Perhaps catch specific exception types if appropriate.  What scenarios trigger an exception?  Consider logging as error instead of debug, if the exception should be exposed to the user.